chapter2/testing.md

```diff
@@ -0,0 +1,177 @@
+# Testing
+
+We strongly encourage contributors to test the elements they create.
+We usually advise on creating two types of tests: Unit Tests and Integration Tests.
+To facilitate writing tests and make the developer experience as smooth as possible, we expose a couple of testing utilities inside `Membrane.Testing` package.
+
+We will discuss and walk you through writing tests on a real-world example for the [AAC decoder](https://github.com/membraneframework/membrane-element-aac) element.
+
+## Setting up
+
+In order to simplify things and make the test easier to maintain and modify, we propose the following solution:
+
+- Create a sample file for the element under test. As an example, for the AAC decoder we'll use a very simple AAC audio file
+- Create a known output as a reference file. For example, we save raw audio frames as the reference output for our AAC Decoder
```

Comment:
You may mention that it's not always a good solution, because most encoders produce nondeterministic output for the same input.